docker-compose-cluster.yml

```diff
@@ -0,0 +1,106 @@
+# Copyright 2024 The ModelarDB Contributors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+version: "3.8"
+
+services:
+  # Object store services.
+  minio-server:
+    image: minio/minio
+    container_name: minio-server
+    ports:
+      - "9000:9000"
+      - "9001:9001"
+    environment:
+      MINIO_ROOT_USER: minioadmin
+      MINIO_ROOT_PASSWORD: minioadmin
+    command: server --console-address ":9001" /data
+
+  create-bucket:
+    image: minio/mc
+    container_name: create-bucket
+    depends_on:
+      - minio-server
+    entrypoint:
+      /bin/sh -c "
+      /usr/bin/mc config host add modelardb-minio http://minio-server:9000 minioadmin minioadmin;
+      /usr/bin/mc mb modelardb-minio/modelardb;
+      exit 0;
+      "
+
+  # Metadata database service.
+  postgres:
+    image: postgres:latest
+    container_name: postgres
+    ports:
+      - "5432:5432"
+    environment:
+      POSTGRES_PASSWORD: modelardb_password
+      POSTGRES_USER: modelardb_user
+      POSTGRES_DB: metadata
+
+  # ModelarDB services.
+  modelardb-manager:
+    image: modelardb
+    build: .
+    container_name: modelardb-manager
+    command: ["target/release/modelardbm", "metadata", "s3://modelardb"]
+    ports:
+      - "9998:9998"
+    environment:
+      METADATA_DB_HOST: postgres
+      METADATA_DB_PASSWORD: modelardb_password
+      METADATA_DB_USER: modelardb_user
+      AWS_ACCESS_KEY_ID: minioadmin
+      AWS_SECRET_ACCESS_KEY: minioadmin
+      AWS_DEFAULT_REGION: eu-central-1
+      AWS_ENDPOINT: http://minio-server:9000
+      AWS_ALLOW_HTTP: "true"
+    depends_on:
+      - minio-server
+      - create-bucket
+      - postgres
+
+  modelardb-edge:
+    image: modelardb
+    container_name: modelardb-edge
+    command: ["target/release/modelardbd", "edge", "data/edge", "grpc://modelardb-manager:9998"]
+    ports:
+      - "9999:9999"
+    environment:
+      MODELARDBD_UNCOMPRESSED_DATA_BUFFER_CAPACITY: 640
```

Comment:
Why overwrite the default settings? If the default settings can be improved I suggest changing them instead.

Reply:
The default settings are more ideal for a production build. The values here are the same as were used in the MORE cluster deployment since they better fit a testing/development build.